configure.ac

```diff
@@ -1310,6 +1310,18 @@ case "$target_os" in
         ;;
 esac
 
+dnl ######################################################################
+dnl Should "processes_select" use "/proc/<nnn>" rather than "ps" command?
+dnl New at March/2019.  Developing on RHEL6.  Eventually aim to use widely.
+dnl ######################################################################
+
+case "$target_os" in
+   linux*)
+        AM_CONDITIONAL([HAVE_PS_VIA_PROC], [test -f "/proc/1/stat"])
```

Comment:
This fails the OS X build in travis:

```
checking that generated files are newer than configure... done
configure: error: conditional "HAVE_PS_VIA_PROC" was never defined.
Usually this means the macro was only invoked conditionally.
The command "./travis-scripts/script.sh" exited with 1.
```

There's many ways to address this, but I think the easiest / most appropriate is to just remove the linux restriction. your `test` command should already check whether `/proc` exists or not.

Reply:
My reason for that temporary working was conservatism while this work is in preparation.  I fully and enthusiastically agree that it should eventually go.

Various OSes also offer "/proc".  But a glance at their man pages (and existing other corners of CFEngine code) suggest that the new code may require some "#ifdef" work before it can go live on such OSes.

If I were to remove that Linux restriction right now, there is a significant chance that new builds on such "/proc"-based non-Linux systems would fail.

So... I'm happy to go with your call.  Do we temporarily keep the Linux restriction and keep the code safe for untested /proc OSes?  Or do we remove the restriction and risk failures on such OSes?

Reply:
Oh!... Could I simply add a default '`*)`' case for this interim period?  Like:

> case "$target_os" in
>    linux*)
>         AM_CONDITIONAL([HAVE_PS_VIA_PROC], [test -f "/proc/1/stat"])
>         ;;
>    *)
>         AM_CONDITIONAL([HAVE_PS_VIA_PROC], false)
>         ;;
> esac

with a comment that the 'case' scaffolding is temporary, to be removed in due course, leaving only the one line "`AM_CONDITIONAL(..., [test...])`"?

Reply:
> Oh!... Could I simply add a default '*)' case for this interim period? Like:

@tdlee Yes, I believe that is a way to do it. It essentially needs to be defined for  all branches (all systems) as either true or false. There might be better / cleaner ways to to do it, but my knowledge of m4/autotools/sh is limited.

Reply:
That is now place at commit '0e49619a' below, including a comment that the 'case' scaffolding is a transitional arrangement.

Reply:
I'm still turning over ideas to make this unsatifactory scaffolding into something cleaner!

At the heart of the matter are:

- the "/proc" implementations seems to vary significantly across systems

- we don't yet have even embryonic implementations for any of the non-Linux systems

- there may be OSes which don't use "/proc" but do offer a neat programming interface, and which it would be good for our autoconf-etc. structure to encompass

Reply:
See commit 'aedbcda' below.  This gives a cleaner framework, capable of detecting whether or not the platform has a coded handler for its "/proc".